tests/formats/light_client/data_collection.md

```diff
@@ -0,0 +1,76 @@
+# Light client data collection tests
+
+This series of tests provides reference test vectors for validating that a full node collects canonical data for serving to light clients implementing the light client sync protocol to sync to the latest block header.
+
+## Test case format
+
+### `initial_state.ssz_snappy`
+
+An SSZ-snappy encoded object of type `BeaconState` to initialize the blockchain from. The state's `slot` is epoch aligned.
+
+### `steps.yaml`
+
+The steps to execute in sequence.
+
+#### `new_block` execution step
+
+The new block described by the test step should be imported, but does not become head yet.
+
+```yaml
+{
+    fork_digest: string                -- encoded `ForkDigest`-context of `block`
+    data: string                       -- name of the `*.ssz_snappy` file to load
+                                          as a `SignedBeaconBlock` object
+}
+```
+
+#### `new_head` execution step
+
+The given block (previously imported) should become head, leading to potential updates to:
+
+- The best `LightClientUpdate` for non-finalized sync committee periods.
+- The latest `LightClientFinalityUpdate` and `LightClientOptimisticUpdate`.
+- The latest finalized `Checkpoint` (across all branches).
+- The available `LightClientBootstrap` instances for newly finalized `Checkpoint`s.
+
+```yaml
+{
+    head_block_root: Bytes32           -- string, hex encoded, with 0x prefix
+    checks: {
```

Comment:
what do you think about aligning with fork choice test format more?

1. Add "Checks step" description.
2. Make `head_block_root` as one of the checking items.

Reply:
The `head_block_root` is the actual operation that is done here (that's the new head to select, and the checks describe the result of updating the head).